### DISCUSSION TITLE
Styles api on Mantine v6 Modal

### DISCUSSION THREAD

**Discussion by FelixBlaisThon** [mantinedev/mantine#3708] [Q&A] [answered]
Hello, 

I've been trying to add custom style to the Modal component from the latest major release 6.0.0 and I've been struggling a lot. 

First I've noticed that `theme.fn.size(...)` was not available anymore but I couldn't see it in the changelog, now it seems that `getSize(...)` is used instead (https://github.com/mantinedev/mantine/commit/97be8534ac35af33b628271ba16e683115d01625). I've tried it but I can't make the width of my modal any different.. 😞 

Here is my sandbox from which I tried to make that custom style. 
https://codesandbox.io/s/mantine-playground-forked-lu3o3y?file=/src/App.tsx:331-625

FYI, if I pass the prop `size={800}` to the `<Modal />` component, it does make the modal bigger with a 800px width, but if I go through styles api and getSize like the snippet below, it does apply the width to the `.mantine-Modal-content`, but the width keeps the default value that Mantine provides.

```
theme={{
        components: {
          Modal: {
            styles: (theme, { size }: ModalBaseProps) => ({
              content: {
                width: getSize({
                  size: size!,
                  sizes: { xs: 440, sm: 550, md: 800, lg: 1334, xl: "85%" }
                }),
               ...
        }
```

Thank you

**Comment by FelixBlaisThon** [comment, marked as answer]
I found later yesterday that it is done with `flex` like so (from https://github.com/mantinedev/mantine/blob/master/src/mantine-core/src/Modal/ModalRoot/ModalRoot.styles.ts#L21)
![image](https://user-images.githubusercontent.com/52010042/223724738-3af4f526-7343-4b70-8124-76ba56d0816e.png)

**Comment by imoby** [comment]
I have a similar issue where I used to us `xs` to style my modals and drawers but now all of that `xs` functionality is gone and broke a lot of my custom modals/drawers. Any reason why `xs` is no longer an option
